File: NordcloudAssignmentHttpTrigger/device.ts
import { LinkStation } from "./linkStations";

export class Device {
	x: number;
	y: number;

	constructor(x: number, y: number) {
		this.x = x;
		this.y = y;
	}

	getLinkStationWithMostPower(linkStations: LinkStation[]): LinkStation {
		let maxLinkStation: LinkStation = null;
		let maxLinkStationPower = 0;

		for (const linkStation of linkStations) {
			const linkStationPower = linkStation.getPower(this)

			if (linkStationPower > maxLinkStationPower) {
				maxLinkStationPower = linkStationPower;
				maxLinkStation = linkStation;
			}
		}

		return maxLinkStation
	}
}

File: v1_python_version/device.py
from link_station import LinkStation

class Device:
	def __init__(self, x, y):
		self.x = x
		self.y = y

	def get_link_station_with_most_power(self, link_stations):
		max_link_station = None
		max_link_station_power = 0

		link_station: LinkStation
		for link_station in link_stations:
			device = self
			link_station_power = link_station.getPower(device) 

			if link_station_power > max_link_station_power:
				max_link_station_power = link_station_power
				max_link_station: LinkStation = link_station

		return max_link_station